### DISCUSSION TITLE
Adding a distractor task

### DISCUSSION THREAD

**Discussion by melisa-akan** [jspsych/jsPsych#1038] [Q&A] [answered]
Hello, 

I would like to add a distractor task in between the different phases of my experiment. I was wondering whether there already exists jsPsych plugins for fun(-ish) tasks that could be set to run for a fixed duration. 

In one of the tutorials, I remember Josh de Leeuw mentioning amazing people in his lab who create video games as plugins.  Of course, my distractor task doesn't need to be as fancy. 

If such plugins do not exist... I see that it is possible to find java script code for games like Tetris online.  Any ideas on whether it is possible to integrate such code to a program written with jsPsych?

If not, does anyone have any suggestions on how to create a simple distractor task using the available plugins?

Thank you so much for any guidance or ideas!

(I don't know how to thank Josh de Leeuw. I don't know what I would do if it weren't for jsPsych. Thank you so much for all your work and for providing so much support.)

Melisa

**Comment by jodeleeuw** [comment]
Hi Melisa.

Thanks so much for the kinds words!

> If such plugins do not exist... I see that it is possible to find java script code for games like Tetris online. Any ideas on whether it is possible to integrate such code to a program written with jsPsych?

In theory this is totally possible. Depending on how the code is written it may be relatively easy or it could be challenging. 

> If not, does anyone have any suggestions on how to create a simple distractor task using the available plugins?

I think this depends on what counts as a distractor task in your experiment. You could use the `vsl-grid-scene` plugin to generate a "spot the change" kind of game, but maybe that's not the right kind of distraction for your experiment. If you can generate a list of candidate distractor tasks, then I could help think about ways to implement them within the scope of the existing plugins.

**Reply by melisa-akan** [reply]
Hi Josh, 

Thank you so much for your response! I am sorry for not being clear earlier. My experiment will have faces and outside scenes. So the distractor task can be anything that does not include these images. I also don't want it to be too boring like solving addition problems as the distractor task will be 5-min long. 

The vsl-grid-scene plugin sounds like a great idea. I am having difficulty figuring out how I could go about creating same vs. different trials, and give feedback.  However, this is a great starting point. Thank you!

A follow-up question: Do you think it would be possible to have the task run for a particular duration and then, move on to the next thing in the timeline when the time is over?

Thanks again for all the help! I really appreciate it.

**Comment by becky-gilbert** [comment, marked as answer]
Hi @melisa-akan! 

You could check out the [Experiment Factory](http://www.expfactory.org/experiments), which has a large set of tasks/games built with jsPsych. Many of their tasks use custom plugins, so you would also need those files, and you might need to remove some of the code that is specific to running the tasks from within their Experiment Factory system.

This [JATOS example studies site](http://www.jatos.org/Example-Studies.html) also has a few freely-available tasks/games - some written in jsPsych and some not. They are all written to run on a JATOS server but could easily be modified to run on a different server.

**Reply by melisa-akan** [reply]
Thanks so much @becky-gilbert!  This is really helpful.
